backend/file-system/src/main/java/com/htec/filesystem/service/ShelfService.java

```diff
@@ -38,4 +52,31 @@ public void createShelf(CreateShelfRequestModel createShelfRequestModel, Long us
 
         shelfRepository.save(shelfEntity);
     }
+
+    @Transactional
+    public void softDeleteShelf(AuthUser user, Long shelfId) {
+
+        ShelfEntity shelfEntity = shelfRepository.findById(shelfId)
+                .orElseThrow(ExceptionSupplier.shelfNotFound);
+
+        List<FolderEntity> folderEntities = folderRepository.findAllByShelfId(shelfId);
+        List<FileEntity> fileEntities = fileRepository.findAllByShelfId(shelfId);
+
+        if (Objects.equals(shelfEntity.getUserId(), user.getId())) {
```

Comment:
maybe return error if this is not equals

Reply:
Done